### PR TITLE
Added latestBlocks, latestBlockIds methods and tests

### PR DESCRIPTION
**Pull request by jingsting**
## Purpose
We could benefit from RPC methods that retrieve a number of blocks and block ids at the head of the chain

## Approach
- Added RPC method `topl_latestBlocks`
- Added RPC method `topl_latestBlockIds`

## Testing
- Added test for both methods

## Tickets
#1843

### CODE REVIEW

**Comment by jingsting** [comment]
- Separated some handler checks for height into more specific ones with error response
- Changed some of the RPC method implementations to use for comprehension and moved some procedures out from the inside of withNodeView method calls
- Added RPC error tests for securityRoot and assetCode

File: node/src/test/scala/co/topl/api/transaction/TransferRPCTestMethods.scala [hunk 1]
```diff
@@ -201,7 +202,7 @@ trait TransferRPCTestMethods extends AnyWordSpec with Matchers with RPCMockState
       |      "assetCode" : "${assetCode.toString}",
```
Comment (b-vennes) [inline node/src/test/scala/co/topl/api/transaction/TransferRPCTestMethods.scala:202]: The `.toString` can be removed now that asset code is a `String`.

File: node/src/main/scala/co/topl/rpc/handlers/NodeViewRpcHandlerImpls.scala [hunk 2]
```diff
@@ -78,28 +80,54 @@ class NodeViewRpcHandlerImpls(
...
+          checkHeightRange(headHeight, params.startHeight, params.endHeight, rpcSettings.blockIdRetrievalLimit)
+        )
+        ids <- withNodeView(view => getBlockIdsInRange(view.history, range._1, range._2))
+      } yield ids
+
+  override val latestBlocks: ToplRpc.NodeView.LatestBlocks.rpc.ServerHandler =
+    params =>
+      for {
+        headHeight <- withNodeView(_.history.height)
+        startHeight = headHeight - params.numberOfBlocks + 1
```
Comment (b-vennes) [inline node/src/main/scala/co/topl/rpc/handlers/NodeViewRpcHandlerImpls.scala:114]: It might be nice to add a comment here for why the `+ 1` is necessary.

Comment (scasplte2) [reply node/src/main/scala/co/topl/rpc/handlers/NodeViewRpcHandlerImpls.scala:114]: were you planning to add this comment @jingsting ?

File: node/src/test/scala/co/topl/api/transaction/TransferRPCTestMethods.scala (hunk 1 above)
Comment (scasplte2) [reply node/src/test/scala/co/topl/api/transaction/TransferRPCTestMethods.scala:202]: please address @jingsting

File: node/src/main/scala/co/topl/rpc/handlers/NodeViewRpcHandlerImpls.scala (hunk 2 above)
Comment (jingsting) [reply node/src/main/scala/co/topl/rpc/handlers/NodeViewRpcHandlerImpls.scala:115]: Yeah, just added a comment there

File: node/src/test/scala/co/topl/api/transaction/TransferRPCTestMethods.scala (hunk 1 above)
Comment (jingsting) [reply node/src/test/scala/co/topl/api/transaction/TransferRPCTestMethods.scala]: Removed it 👍

**Pull request merged**